stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/AddChannelView.kt

```diff
@@ -52,6 +52,18 @@ class AddChannelView : FrameLayout {
         init(attrs)
     }
 
+    override fun onInterceptTouchEvent(ev: MotionEvent): Boolean {
+        if (binding.messageInputView.isVisible && ev.action == MotionEvent.ACTION_DOWN) {
+            val messageInputViewRect = Rect().apply {
+                binding.messageInputView.getHitRect(this)
+            }
+            if (messageInputViewRect.contains(ev.x.toInt(), ev.y.toInt())) {
+                controller.messageInputViewClicked()
+            }
+        }
+        return super.onInterceptTouchEvent(ev)
```

Comment:
Couldn't we somehow get this from `MessageInputView` in a cleaner way, with a listener or something?

Reply:
It's not easy. `MessageInputView` has few children which consume on click event and I don't want to override `onInterceptTouchEvent ` inside `MessageInputView` cause I don't think discovering clicks on the whole view will be a common scenario.
I neither like this solution but I didn't find anything better 😞

Reply:
And it's not enough to detect when its input field gets focused, we want all interactions with the view?

Reply:
I think it's not enough because there are edge cases when input field won't gain focus (for example adding attachment).
It doesn't look good when you do that and you won't see channel preview

Reply:
Alright, that makes sense, let's go with this then.